Keda crashes if the deployment doesn't exist

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x108 pc=0x1192bcc]

goroutine 77 [running]:
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).scaleDeployment(0xc000162120, 0x0, 0xc0004bb600, 0x0)
        /go/src/github.com/kedacore/keda/pkg/handler/scale_handler.go:240 +0x2c
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).handleScale(0xc000162120, 0x1e7f760, 0xc0006be400, 0xc0004bb600)
        /go/src/github.com/kedacore/keda/pkg/handler/scale_handler.go:234 +0x1a4
github.com/kedacore/keda/pkg/handler.(*ScaleHandler).handleScaleLoop(0xc000162120, 0x1e7f760, 0xc0006be400, 0xc0004bb600, 0xc0006ca801)
        /go/src/github.com/kedacore/keda/pkg/handler/scale_handler.go:186 +0x6a
created by github.com/kedacore/keda/pkg/handler.(*ScaleHandler).WatchScaledObjectWithContext
        /go/src/github.com/kedacore/keda/pkg/handler/scale_handler.go:53 +0x96
```

--- a/pkg/handler/scale_handler.go
+++ b/pkg/handler/scale_handler.go
@@ -231,7 +231,9 @@ func (h *ScaleHandler) handleScale(ctx context.Context, scaledObject *keda_v1alp
 		scaler.Close()
 	}
 
-	h.scaleDeployment(deployment, scaledObject, isScaledObjectActive)
+	if deployment != nil {
+		h.scaleDeployment(deployment, scaledObject, isScaledObjectActive)
+	}
 
 	return
 }

--- a/pkg/handler/scale_handler.go
+++ b/pkg/handler/scale_handler.go
@@ -216,9 +216,13 @@ func (h *ScaleHandler) handleScaleLoop(ctx context.Context, scaledObject *keda_v
 // handleScale contains the main logic for the ScaleHandler scaling logic.
 // It'll check each trigger active status then call scaleDeployment
 func (h *ScaleHandler) handleScale(ctx context.Context, scaledObject *keda_v1alpha1.ScaledObject) {
+	scalers, deployment := h.getScalers(scaledObject)
+	if deployment == nil {
+		return
+	}
+
 	isScaledObjectActive := false
 
-	scalers, deployment := h.getScalers(scaledObject)
 	for _, scaler := range scalers {
 		isTriggerActive, err := scaler.IsActive(ctx)
 		if err != nil {
@@ -231,9 +235,7 @@ func (h *ScaleHandler) handleScale(ctx context.Context, scaledObject *keda_v1alp
 		scaler.Close()
 	}
 
-	if deployment != nil {
-		h.scaleDeployment(deployment, scaledObject, isScaledObjectActive)
-	}
+	h.scaleDeployment(deployment, scaledObject, isScaledObjectActive)
 
 	return
 }

